readme: adding quickstart section (#1291)
sphinx-gallery/sphinx-gallery: README.rst

   :width: 80%
   :alt: A demo of a gallery generated by Sphinx-Gallery

Quickstart
==========

Sphinx-Gallery can be used to generate an example gallery for a library, as well as a stand-alone web page showcasing examples of a particular Python package, module, or class.

Let's get started with a simple example or checkout the `documentation <sphinx-gallery.github.io>`_ for introductions on how to use it and more...

Install via ``pip``
-------------------

.. installation-begin-content

You can do a direct install via ``pip`` by using:

.. code-block:: bash

    $ pip install sphinx-gallery "sphinx>=4.0" pillow

.. warning::
    Sphinx-Gallery will not manage its dependencies when installing, thus
    you are required to install them manually. Our minimal dependencies
    are **Sphinx >= 4** and Pillow, which we use for scaling images.

.. note::
    Sphinx-Gallery has also support for scraping images from packages like

>     Matplotlib, Seaborn, and PyVista. We recommend installing system ``optipng``

Reviewer: could we clarify that we support matplotlib and matplotlib based packages such as Seaborn and PyVista?

Author: well, this is taken from the actual Readme, so do you have suggestions on how to improve it?

Reviewer: Ah good point. Our [image scraper doc](https://sphinx-gallery.github.io/stable/configuration.html#image-scrapers) says:

> The default value is 'image_scrapers': ('matplotlib',) which only scrapes Matplotlib images. Note that this includes any images produced by packages that are based on Matplotlib, for example Seaborn or Yellowbrick.

What about:

"Sphinx-Gallery has also support for scraping images from Matplotlib and Matplotlib based packages such as Seaborn, and PyVista."

Author: updated

    binaries to reduce the file sizes of the generated PNG files.

.. installation-end-content

Add examples to your docs
-------------------------

Let's assume simple scenario, you have a Python package with a directory structure like this:

.. code-block::
    .
    ├── doc
    │   ├── conf.py
    │   ├── index.rst
    |   ├── make.bat
    │   └── Makefile
    ├── my_python_module
    │   ├── __init__.py
    │   └── mod.py
    └── examples
        ├── plot_example.py
        └── README.txt (or .rst)

Then you just add the following to your ``doc/conf.py``:

.. code-block:: python

    extensions = [
        ...
        'sphinx_gallery.gen_gallery',
    ]

    # path to the examples scripts
    sphinx_gallery_conf = {
        'examples_dirs': '../examples',   # path to your example scripts
        'gallery_dirs': 'gallery',  # path to where to save gallery generated output
    }

> Finally just compile your docs as usual. You will see a new section "gallery" in your docs with the examples.

Reviewer: I think you need to add the gallery index (e.g., `examples/index.rst`) file to the main `doc/index.rst` file for it to appear in the docs ?

Author: yes that is good addition

   :width: 80%
   :alt: A demo of a gallery generated by Sphinx-Gallery

Quickstart
==========

Sphinx-Gallery can be used to generate an example gallery for a library,
 as well as a stand-alone web page showcasing examples of a particular
 Python package, module, or class.

Let's get started with a simple example or checkout the
 `documentation <sphinx-gallery.github.io>`_ for introductions on how
 to use it and more...

Install via ``pip``
-------------------

.. installation-begin-content

You can do a direct install via ``pip`` by using:

.. code-block:: bash

    $ pip install sphinx-gallery "sphinx>=4.0" pillow

.. warning::
    Sphinx-Gallery will not manage its dependencies when installing, thus
    you are required to install them manually. Our minimal dependencies
    are **Sphinx >= 4** and Pillow, which we use for scaling images.

.. note::

>     Sphinx-Gallery has also support for scraping images from packages like
>     Matplotlib, Seaborn, and PyVista. We recommend installing system ``optipng``

Author: ```suggestion
    Sphinx-Gallery has also support for scraping images from Matplotlib and Matplotlib based
    packages such as Seaborn, and PyVista. We recommend installing system ``optipng``
```

>     Sphinx-Gallery has also support for scraping images from Matplotlib and Matplotlib based
>     packages such as Seaborn, and PyVista. We recommend installing system ``optipng``
>     binaries to reduce the file sizes of the generated PNG files.

Author: ```suggestion
    Sphinx-Gallery has also support for scraping images from Matplotlib
    and Matplotlib based packages such as Seaborn, and PyVista.
    We recommend installing system ``optipng`` binaries to reduce
    the file sizes of the generated PNG files.
```

   :width: 80%
   :alt: A demo of a gallery generated by Sphinx-Gallery

Quickstart
==========

> Sphinx-Gallery can be used to generate an example gallery for a library,

Reviewer: ```suggestion
Sphinx-Gallery can be used to generate an example gallery from `.py` files
for a library,
```

 as well as a stand-alone web page showcasing examples of a particular
 Python package, module, or class.

Let's get started with a simple example or checkout the
 `documentation <sphinx-gallery.github.io>`_ for introductions on how
 to use it and more...

Install via ``pip``
-------------------

.. installation-begin-content

You can do a direct install via ``pip`` by using:

.. code-block:: bash

    $ pip install sphinx-gallery "sphinx>=4.0" pillow

.. warning::
    Sphinx-Gallery will not manage its dependencies when installing, thus
    you are required to install them manually. Our minimal dependencies
    are **Sphinx >= 4** and Pillow, which we use for scaling images.

.. note::
    Sphinx-Gallery also has support for scraping images from Matplotlib

>     and Matplotlib-based packages such as Seaborn and PyVista.

Reviewer 2: ```suggestion
    and Matplotlib-based packages such as Seaborn.
```

Reviewer: I questioned this too. When did we support PyVista?

    We recommend installing system ``optipng`` binaries to reduce
    the file sizes of the generated PNG files.

.. installation-end-content

Add examples to your docs
-------------------------

Let's assume simple scenario, you have a Python package with a directory structure like this:

.. code-block::

    ├── doc
    │   ├── conf.py
    │   ├── index.rst
    |   ├── make.bat
    │   └── Makefile
    ├── my_python_module
    │   ├── __init__.py
    │   └── mod.py
    └── examples
        ├── plot_example.py
        └── README.txt (or .rst)

> Then you just add the following to your ``doc/conf.py``:

Reviewer: ```suggestion
Enable Sphinx-Gallery by adding the following to your ``doc/conf.py``:
```
nit

.. code-block:: python

    extensions = [
        ...
        'sphinx_gallery.gen_gallery',
    ]

    # path to the examples scripts
    sphinx_gallery_conf = {
        'examples_dirs': '../examples',   # path to your example scripts

>         'gallery_dirs': 'gallery',  # path to where to save gallery generated output

Reviewer: ```suggestion
        'gallery_dirs': 'auto_examples',  # path to where to save gallery generated output
```
Lets use 'auto_examples' because that's what we use elsewhere.

    }

Finally just compile your docs as usual.

>  You will find a new page "gallery" in your docs with the examples.

Reviewer: ```suggestion
Sphinx-Gallery will generated reST files, adding execution outputs, and save them in `auto_examples/`.
```

>  To get it linked from the main page, add a link ``gallery/index.rst``

Reviewer: ```suggestion
Add a link to ``auto_examples/index.rst`` to include the gallery in your documentation.
```

They can include the gallery anywhere they want really

   :width: 80%
   :alt: A demo of a gallery generated by Sphinx-Gallery

Quickstart
==========

> Sphinx-Gallery can be used to generate an example gallery from `.py` files

Reviewer 2: reST single backtick is autolink, double-backticks are code. You want code here I think
```suggestion
Sphinx-Gallery can be used to generate an example gallery from ``.py`` files
```

> Sphinx-Gallery can be used to generate an example gallery from ``.py`` files

Reviewer: ```suggestion
Sphinx-Gallery can be used to generate an example gallery from ``.py`` files,
```

.. note::
    Sphinx-Gallery also has support for scraping images from Matplotlib
    and Matplotlib-based packages such as Seaborn.
    We recommend installing system ``optipng`` binaries to reduce
    the file sizes of the generated PNG files.

.. installation-end-content

Add examples to your docs
-------------------------

Let's assume simple scenario, you have a Python package with a directory structure like this:

.. code-block::

    ├── doc
    │   ├── conf.py
    │   ├── index.rst
    |   ├── make.bat
    │   └── Makefile
    ├── my_python_module
    │   ├── __init__.py
    │   └── mod.py
    └── examples
        ├── plot_example.py
        └── README.txt (or .rst)

Enable Sphinx-Gallery by adding the following to your ``doc/conf.py``:

.. code-block:: python

    extensions = [
        ...
        'sphinx_gallery.gen_gallery',
    ]

    # path to the examples scripts
    sphinx_gallery_conf = {
        'examples_dirs': '../examples',   # path to your example scripts
        'gallery_dirs': 'auto_examples',  # path to where to save gallery generated output
    }

Finally just compile your docs as usual.

> Sphinx-Gallery will generated reST files, adding execution outputs, and save them in `auto_examples/`.

Reviewer 2: ```suggestion
Sphinx-Gallery will generated reST files, adding execution outputs, and save them in ``auto_examples/``.
```

> Sphinx-Gallery will generated reST files, adding execution outputs, and save them in ``auto_examples/``.

Reviewer: ```suggestion
Sphinx-Gallery will generate reST files, adding execution outputs, and save them in ``auto_examples/``.
```

Add a link to ``auto_examples/index.rst`` to include the gallery in your documentation.

> in your ``doc/index.rst``.

Reviewer: ```suggestion

```
